packages/forms/src/UIForm/UIForm.component.js

```diff
@@ -119,28 +124,13 @@ export default class UIForm extends React.Component {
 	 * trigger The type of trigger
 	 * schema The field schema
 	 */
-	onTrigger(event, payload) {
-		const { formName, updateForm, onTrigger, setError, properties } = this.props;
+	onTrigger(formData, formId, propertyName, propertyValue) {
+		const { onTrigger } = this.props;
 		if (!onTrigger) {
 			return null;
 		}
 
-		return onTrigger(
-			event,
-			{
-				formName,
-				properties,
-				...payload,
-			}
-		)
-			.then(newForm => updateForm(
-				formName,
-				newForm.jsonSchema,
-				newForm.uiSchema,
-				newForm.properties,
-				newForm.errors)
-			)
-			.catch(({ errors }) => setError(formName, errors));
+		return onTrigger(formData, formId, propertyName, propertyValue);
```

Comment:
Where is the part that update the form if we receive a new form from trigger in connector ?
I see it in container that manage internal state, but not in the connected one anymore. Does it mean that user has to manage it himself ?

Reply:
it's react so one directional flow
trigger is an action managed by the parent.
And schema is passed by the parent.
So it's the parent that must handle the update of the schema

Reply:
yes, that's something I was torn appart, but what I meant here is that you moved that on UIForms.container (the react state one), but not in UIForms.connect.js (the redux one)